var/spack/repos/builtin/packages/sundials/package.py

```diff
@@ -27,138 +27,531 @@
 import sys
 
 
-class Sundials(Package):
+class Sundials(CMakePackage):
     """SUNDIALS (SUite of Nonlinear and DIfferential/ALgebraic equation
     Solvers)"""
 
-    homepage = "http://computation.llnl.gov/casc/sundials/"
-    url = "http://computation.llnl.gov/projects/sundials-suite-nonlinear-differential-algebraic-equation-solvers/download/sundials-2.6.2.tar.gz"
+    homepage = "https://computation.llnl.gov/projects/sundials"
+    url = "https://computation.llnl.gov/projects/sundials/download/sundials-2.7.0.tar.gz"
 
-    version('2.7.0', 'c304631b9bc82877d7b0e9f4d4fd94d3')
+    # ==========================================================================
+    # Versions
+    # ==========================================================================
+
+    version('3.0.0-beta-2',
+            git='https://github.com/LLNL/sundials.git', tag='v3.0.0-beta-2')
+    version('2.7.0', 'c304631b9bc82877d7b0e9f4d4fd94d3', preferred=True)
     version('2.6.2', '3deeb0ede9f514184c6bd83ecab77d95')
 
+    # ==========================================================================
+    # Variants
+    # ==========================================================================
+
+    # SUNDIALS solvers
+    variant('cvode',  default=True,
+            description='Enable CVODE')
+    variant('cvodes', default=True,
+            description='Enable CVODES')
+    variant('arkode', default=True,
+            description='Enable ARKode')
+    variant('ida',    default=True,
+            description='Enable IDA')
+    variant('idas',   default=True,
+            description='Enable IDAS')
+    variant('kinsol', default=True,
+            description='Enable KINSOL')
```

Comment:
Can you turn them in an array and loop over them? (see `lammps` for an example)

Reply:
Thanks for the suggestion and the pointer to `lammps`. I've updated these options to an array of `sundials` solvers.